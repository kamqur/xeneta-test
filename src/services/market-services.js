import axios from "axios";

const getMarketRates = (origin, destination) =>
  axios.get(
    `https://685rp9jkj1.execute-api.eu-west-1.amazonaws.com/prod/rates?origin=${origin}&destination=${destination}`,
    {
      headers: {
        "x-api-key": process.env.REACT_APP_API_SECRET_KEY
      }
    }
  );

export { getMarketRates };
