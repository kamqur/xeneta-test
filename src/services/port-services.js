import axios from "axios";

const getPorts = () =>
  axios.get(
    `https://685rp9jkj1.execute-api.eu-west-1.amazonaws.com/prod/ports`,
    {
      headers: {
        "x-api-key": process.env.REACT_APP_API_SECRET_KEY
      }
    }
  );

export { getPorts };
