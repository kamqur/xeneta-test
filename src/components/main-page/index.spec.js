import React from "react";
import { shallow, mount } from "enzyme";

import MainPage from "./index";
import * as api from "../../services/port-services";

describe("MainPage component testing", () => {
  it("snapshort testing", () => {
    const component = mount(<MainPage />);
    expect(component).toMatchSnapshot();
  });

  it("when data is return", () => {
    const mockGetData = jest
      .spyOn(api, "getPorts")
      .mockReturnValue(Promise.resolve([]));
    const component = shallow(<MainPage />);
  });
});
