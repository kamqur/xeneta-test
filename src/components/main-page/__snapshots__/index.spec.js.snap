// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component testing snapshort testing 1`] = `
<MainPage>
  <div
    className="App"
  >
    <div>
       Loading ports ... 
    </div>
  </div>
</MainPage>
`;
